### PR TITLE
CORE-32732 Improved privacy logging.

### PR DESCRIPTION
**Pull request by Aaronius**
## Description
Improves privacy logging so customers better know how opt-in is tied to command queuing and failure.

## Related Issue
https://jira.corp.adobe.com/browse/CORE-32732

## Motivation and Context
Give users a better sense of why their commands are being delayed.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All tests pass and I've made any necessary test changes.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: src/core/createOptIn.js [hunk 1]
```diff
@@ -47,12 +47,17 @@ export default () => {
...
+     * @param {Object} logger A logger object.
+     * @param {Object} _cookie A cookie management object.
      * to the Privacy component.
      */
-    enable(_cookie) {
+    enable(logger, _cookie) {
       cookie = _cookie;
       purposes = cookie.get(COOKIE_NAMESPACE) || PENDING;
+
+      if (purposes === PENDING) {
```
Comment (artur-ciocanu) [inline src/core/createOptIn.js]: I would extract the message into constant, so you could reference it in different places.

File: test/unit/specs/core/createOptIn.spec.js [hunk 2]
```diff
@@ -55,8 +57,11 @@ describe("createOptIn", () => {
 
     it("considers the user pending opt in if cookie is not set", () => {
       const optedInSpy = jasmine.createSpy();
-      optIn.enable(cookie);
+      optIn.enable(logger, cookie);
 
+      expect(logger.warn).toHaveBeenCalledWith(
```
Comment (artur-ciocanu) [inline test/unit/specs/core/createOptIn.spec.js]: Same here

File: src/core/createOptIn.js (hunk 1 above)
Comment (Aaronius) [reply src/core/createOptIn.js]: @artur-ciocanu Are you just talking about putting `"Some commands may be delayed until the user opts in."` as a constant at the top of the file, or are you saying expose it in such a way that other modules or the test can also reference it? Are you suggesting we put all log messages in constants?

**Comment by Aaronius** [comment]
@artur-ciocanu I'm going to merge this, but I'd be happy to open a new PR after you respond to my questions. Thanks.

**Pull request merged**